### ISSUE TITLE
跑unet时，训练可以进行，但是验证过程中崩溃，崩溃原因是内存不够用？有什么解决方法吗？

### ISSUE THREAD

**Issue by zsffuture** [PaddlePaddle/PaddleSeg#923]
如题。
![image](https://user-images.githubusercontent.com/39027951/112791172-81f76b00-9093-11eb-9093-9f1ea3a2d813.png)
![image](https://user-images.githubusercontent.com/39027951/112791191-8e7bc380-9093-11eb-8b05-6146a3e4bb64.png)

**Comment by zsffuture** [comment]
我已经把batch设置为1了，显存是11G的，同时推理时也是内存不够用而崩溃，什么原因呢？

**Comment by zsffuture** [comment]
这里如何有出现验证和推理很占GPU情况时，建议使用delevop版本，同时不要安装pip方式的即pip install paddleseg，我不知道是哪个影响的，我的解决方案是使用delevop版本，二是卸载了paddleseg，之前使用pip install paddleseg，同时batch均设置为1，这样才避免在训练时验证崩溃的情况 。

**Comment by BeyondYourself** [comment]
@zsffuture   这个我也经常遇到过，感觉好像有之前的进程一直在占用

**Comment by BeyondYourself** [comment]
@zsffuture   这个我也经常遇到过，感觉好像有之前的进程一直在占用

**Comment by wuyefeilin** [comment]
看情况，是在验证的时候输入图片过大导致的内存不足，验证一下是否是val_dataset里transform的配置输出的图像大小过大。另外给一下配置让我们确认一下

**Comment by BeyondYourself** [comment]
之前我的总数据量大概是几百兆的时候也会出现这样的情况，而且总的数据量也就几百的情况下也会出现内存不足

**Comment by wuyefeilin** [comment]
> 之前我的总数据量大概是几百兆的时候也会出现这样的情

跟数据量没有关系的，跟transform的设置有关

**Comment by BeyondYourself** [comment]
我之前val中的transform,只有resize  [512,512]这样的设置或者normalize，没有其他别的设置

> > 之前我的总数据量大概是几百兆的时候也会出现这样的情
> 
> 跟数据量没有关系的，跟transform的设置有关

**Comment by wuyefeilin** [comment]
> 我之前val中的transform,只有resize [512,512]这样的设置或者normalize，没有其他别的设置
> 
> > > 之前我的总数据量大概是几百兆的时候也会出现这样的情
> > 
> > 
> > 跟数据量没有关系的，跟transform的设置有关

麻烦给一下你出问题时候的配置信息

**Comment by BeyondYourself** [comment]
这个是我的配置信息
`train_dataset:
  type: Dataset
  dataset_root: /home/aistudio/work/data/tables/label
  train_path: /home/aistudio/work/data/tables/label/train_list.txt
  num_classes: 5
  transforms:  
    - type: RandomHorizontalFlip
    - type: RandomVerticalFlip
    - type: Normalize
    - type: RandomBlur
      prob: 0.23
    - type: Resize 
  mode: train

val_dataset:
  type: Dataset
  dataset_root:  /home/aistudio/work/data/tables/label
  val_path:  /home/aistudio/work/data/tables/label/val_list.txt
  num_classes: 5
  transforms:
    - type: Resize
  mode: val

learning_rate:
  value: 0.001
  decay:
    type: poly
    power: 0.9
    end_lr: 0.0

optimizer:
  type: adam

loss: 
  types:
    - type: CrossEntropyLoss
  coef: [1] `

**Comment by wuyefeilin** [comment]
看一下显存是否只有这一个进程，另外看一下单独评估是否还是一样， 还有缩小Resize的尺寸试一下

**Comment by YRDTSS** [comment]
您好请问您的这个问题解决了吗，我也遇到类似的问题。第一次评估完模型就停止训练了。我自己的数据集两很少，为啥也会出现这种情况呢，是我的transforms的策略的问题吗？求指教
